src/collection_functions.php

```diff
@@ -1390,7 +1390,7 @@ function ($collection) {
  * Transpose each item in a collection, interchanging the row and column indexes.
  * Can only transpose collections of collections. Otherwise an InvalidArgument is raised.
  *
- * @param Collection $collection
+ * @param iterable<Collection> $collection
```

Comment:
It seems that Scrutinizer is not able to parse `iterable<Collection>` doc blocks. Even if the [docs for arrays](https://scrutinizer-ci.com/docs/tools/php/php-analyzer/guides/annotating_code#arrays) say:

> we encourage to specify a more specific type via comments using either the Type[] syntax, or the array&lt;Type&gt; syntax; both are equivalent.

So maybe we should type hint it as `Collection[]`? I'm not sure if this would be correct, though...